File: firefly-example/src/main/java/com/firefly/example/reactive/coffee/store/model/Order.java
package com.firefly.example.reactive.coffee.store.model;

import com.firefly.db.annotation.Column;
import com.firefly.db.annotation.Id;
import com.firefly.db.annotation.Table;

import java.util.Date;

/**
 * @author Pengtao Qiu
 */
@Table(value = "order", catalog = "coffee_store")
public class Order {

    @Id("id")
    private Long id;

    @Column("status")
    private Integer status;

    @Column("amount")
    private Long amount;

    @Column("price")
    private Double price;

    @Column("total_price")
    private Double totalPrice;

    @Column("product_id")
    private Long productId;

    @Column("user_id")
    private Long userId;

    @Column("description")
    private String description;

    @Column("create_time")
    private Date createTime;

    @Column("update_time")
    private Date updateTime;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Integer getStatus() {
        return status;
    }

    public void setStatus(Integer status) {
        this.status = status;
    }

    public Long getAmount() {
        return amount;
    }

    public void setAmount(Long amount) {
        this.amount = amount;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public Double getTotalPrice() {
        return totalPrice;
    }

    public void setTotalPrice(Double totalPrice) {
        this.totalPrice = totalPrice;
    }

    public Long getProductId() {
        return productId;
    }

    public void setProductId(Long productId) {
        this.productId = productId;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getCreateTime() {
        return createTime;
    }

    public void setCreateTime(Date createTime) {
        this.createTime = createTime;
    }

    public Date getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(Date updateTime) {
        this.updateTime = updateTime;
    }

}

File: firefly-example/src/main/kotlin/com/firefly/example/kotlin/coffee/store/model/Order.kt
package com.firefly.example.kotlin.coffee.store.model

import com.firefly.db.annotation.Column
import com.firefly.db.annotation.Id
import com.firefly.db.annotation.Table
import java.util.*

/**
 * @author Pengtao Qiu
 */
@Table(value = "order", catalog = "coffee_store")
data class Order(@Id("id") var id: Long,
                 @Column("status") var status: Int,
                 @Column("amount") var amount: Long,
                 @Column("price") var price: Double,
                 @Column("total_price") var totalPrice: Double,
                 @Column("product_id") var productId: Long,
                 @Column("user_id") var userId: Long,
                 @Column("description") var description: String,
                 @Column("create_time") var createTime: Date,
                 @Column("update_time") var updateTime: Date) {

    override fun equals(other: Any?): Boolean = if (other is Order) Objects.equals(id, other.id) else false

    override fun hashCode(): Int = Objects.hashCode(id)
}